ConfigSpace/conditions.py

```diff
@@ -176,6 +209,25 @@ def evaluate(self, instantiated_hyperparameters: Hyperparameter) -> bool:
 
         return self._evaluate(evaluations)
 
+    def evaluate_vector(self, instantiated_vector: List[int]) -> bool:
+        # Then, check if all parents were passed
+        conditions = self.get_descendant_literal_conditions()
+        for condition in conditions:
+            if condition.parent_vector_id not in range(len(instantiated_vector)):
+                raise ValueError("Evaluate must be called with all "
```

Comment:
Do you want to keep this check, because the instantiated vector is always the full configuration vector? It can happen that the vector parent id is None if the condition was not called to set it's vector id on time (however, this should not happen in the current implementation), but that is easier to check.